packages/eslint-plugin-typescript/src/rules/json-util.ts

```diff
@@ -0,0 +1,153 @@
+import { Rule } from 'eslint';
+import type { Literal, Node, ObjectExpression, Program, Property, SpreadElement } from 'estree';
+import { JsonPropertyAssertion } from './json-property-assertion';
+
+/**
+ * Create a new rule validating a property value of a given json file.
+ * @param context ESLint {@link Rule.RuleContext} for reporting issues.
+ * @param propertyAssertion Descriptor of the property and expected value which should be validated.
+ * @param filename The filename of the file being validated.
+ */
+export function jsonRule(
+    context: Rule.RuleContext,
+    propertyAssertion: JsonPropertyAssertion,
+    filename: string
```

Comment:
I think you should rather get the current filename from the context be calling `getFilename()` like described here: https://eslint.org/docs/latest/developer-guide/working-with-rules

That way you don't need this weird hack [here](https://github.com/cloudflightio/cloudflight-eslint-plugin/pull/10/files#diff-f34eef32f09a568343b13a01c70e2e819632649902b02c1e05135f8f3deff3eaR6) which would print an incorrect filename for the `tsconfig` effected

This is true for all methods in this file.

Reply:
I think the variable name is misleading, as it isn't necessarily the file name, but rather more used to further customize the message in case of a violation, as the filename is anyway present in the linter output.
For Example the `compilerOptions` rules we have print a message: `TS Compiler option 'xy' is missing!`, which is not necessarily the filename.

I'll try to come up with a better naming, feel free to suggest something if you have an idea :)

Reply:
Renamed the parameter to `ruleContextName` and added a more clear js-doc description, does that also work for you?